### DISCUSSION TITLE
Geotracking avec IGO2

### DISCUSSION THREAD

**Discussion by solucompme** [infra-geo-ouverte/igo2-quebec#42] [Q&A]
Bonjour ! IGO2 supporte-t-il encore la fonction de géorepérage/suivi de dispositifs dans le style de Traccar, GeoTab, FleetComplete ?

**Comment by pelord** [comment]
@matrottier websocket?

**Comment by pelord** [comment]
Actuellement, il y a un connecteur de données de type websocket. Par contre, il n'y a pas de service démo pour ce format. C'est @matrottier qui la développé il y a quelques années pour le suivi de véhicules.

**Reply by solucompme** [reply]
D'accord, a-t-il de la documentation à ce sujet pour le faire ? Nous désirons implanter notre propre igo2 pour notre service de recherche et sauvetage et y inclure le suivi GPS des gens sur le terrain, combiné aux couches d'informations et fonctions incroyables de IGO2. Merci !

**Reply by matrottier** [reply]
https://igo2.readthedocs.io/fr/latest/properties.html#websocket : 
si ma vieille mémoire ne fait pas défaut, avec un définition comme dans la documentation et un websocket qui retourne un GeoJSON, la magie va opérer d'elle même. 
exemple plus complet, la carte ce déplace pour suivre le ID 19-0020:
``` 
{
      "title": "Point temps réel",
      "visible": true,
      "sourceOptions": {
        "type": "websocket",
        "url": "wss://websocket.domain/apis/websocket/",
        "formatOptions": {
                "dataProjection": "EPSG:4326",
                "featureProjection": "EPSG:3857"
              },
        "onmessage": "update",
        "queryable": true,
        "queryTitle": "Véhicule : ${unitid}",
        "sourceFields": [
          { "name": "unitid", "alias": "ID véhicule", "excludeFromOgcFilters": true },
          { "name": "nom_origine", "alias": "Zone de départ", "excludeFromOgcFilters": true },
          { "name": "date_debut_trajet", "alias": "Date du départ", "excludeFromOgcFilters": true },
          { "name": "speed", "alias": "Vitesse actuelle", "excludeFromOgcFilters": true },
          { "name": "vitesse permise", "alias": "Vitesse permise", "excludeFromOgcFilters": true }
        ]
          },
      "animation":{
        "duration": "2000"
      },
      "trackFeature": "19-0020"
    }
```
c'est une vieille configuration qui date de 3 ans et il y a pas eu de mise a jour car le projet a été arrêté, p-e des petites modifications à apporter. Je n'ai malheureusement plus d'exemple du JSON qui était envoyé par le websocket.

**Reply by matrottier** [reply]
j'ai retrouvé ce que le trigger dans la notre BD Postgres renvoyait : 
```sql

-- rec_result = select sur une table X

select 'Feature' As type, 
  		 rec_result.unitid as id,
  		--projection as projection,
		ST_AsGeoJSON(st_geomfromtext('POINT('||rec_result.long||' '||rec_result.lat||')'))::json as geometry,
		row_to_json(rec_result)::json as properties into rec_geojson;
  
  PERFORM pg_notify('postgresql2websocket', row_to_json(rec_geojson)::text);
  RETURN NEW;

```
